website/pages/blog/cloudquery-github-action.md

```diff
@@ -0,0 +1,152 @@
+---
+title: Running CloudQuery with GitHub Actions
+tag: tutorial
+date: 2022/10/25
+description: >-
+  CloudQuery Deploy-less Orchestration with GitHub Actions
+author: erezrokah
+---
+
+import { BlogHeader } from "../../components/BlogHeader"
+
+<BlogHeader/>
+
+In this blog post, we will show you how to run CloudQuery with [GitHub Actions](https://github.com/features/actions) and the AWS source plugin.
+
+## Pre-requisites
+
+Since we'll be running CloudQuery in the context of a GitHub Action runner, we'll need to add AWS authentication.
+
+To set up authentication with AWS from GitHub Actions you can follow the [Configuring OpenID Connect in Amazon Web Services blog](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) from GitHub.
+
+## Creating the CloudQuery configuration file
+
+Under the root of your repository, create a new `cloudquery.yml` file with the following content:
+
+```yaml
+kind: source
+spec:
+  name: 'aws'
+  path: cloudquery/aws
+  version: 'v3.4.2' # latest version of aws plugin
+  destinations: ['postgresql']
+---
+kind: destination
+spec:
+  name: 'postgresql'
+  path: cloudquery/postgresql
+  version: 'v1.3.3' # latest version of postgresql plugin
+  spec:
+    connection_string: ${CQ_DSN} # The CQ_DSN environment variable will be set by GitHub Action workflow
+```
+
+> For more configuration options, [visit our docs](https://www.cloudquery.io/docs/reference/source-spec)
+
+## Creating the GitHub Action workflow
+
+First we'll need [to create a GitHub secret](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) with the name `CQ_DSN` and the value of the connection string to your PostgreSQL database.
+
+Create a workflow file under `.github/workflows/cloudquery.yml` with the following content:
```

Comment:
Add: "Fill in `<role-arn>` and `<region>` according to the role you created in the prerequisites."